basedtyping/__init__.py

```diff
@@ -45,6 +51,18 @@
 """A value that can never exist. This is the narrowest possible form."""
 
 
+def _type_convert(arg: object) -> object:
+    """For converting None to type(None), and strings to ForwardRef.
+
+    Stolen from typing.
+    """
+    if arg is None:
+        return type(None)
+    if isinstance(arg, str):
+        return ForwardRef(arg)
+    return arg
+
+
```

Comment:
```suggestion
if sys.version_info >= (3, 10):
    from typing import _type_convert
else:
    def _type_convert(arg: object) -> object:
        """For converting None to type(None), and strings to ForwardRef."""
        if arg is None:
            return type(None)
        if isinstance(arg, str):
            return ForwardRef(arg)
        return arg


```

Reply:
maybe we should have a seperate polyfills/downlevel module to avoid bloating the main basedtyping module with copies of stuff from newer python versions

Reply:
util?

Reply:
`_type_convert` isn't typed, so it's just simpler to steal it here. it's really short, so I think it's fine for now to keep as it.